geo/src/algorithm/rotate.rs

```diff
@@ -345,4 +352,43 @@ mod test {
         let line1 = Line::new(Point::new(0., 0.), Point::new(-2., 0.));
         assert_relative_eq!(line0.rotate_around_point(90., Point::new(0., 0.)), line1);
     }
+
+    #[test]
+    fn test_rotate_multipolygon_around_centroid() {
+        let multipolygon: MultiPolygon<f64> = vec![
+            polygon![
+                (x: 0., y: 0.),
+                (x: 10., y: 0.),
+                (x: 10., y: 10.),
+                (x: 0., y: 10.),
+                (x: 0., y: 0.),
+            ], 
+            polygon![
+                (x: 0., y: 0.),
+                (x: -10., y: 0.),
+                (x: -10., y: -10.),
+                (x: 0., y: -10.),
+                (x: 0., y: 0.),
+            ]
+        ].into();
+
+        let expected: MultiPolygon<f64> = vec![
+            polygon![
+                (x: 0., y: 0.),
+                (x: 10., y: 0.),
+                (x: 10., y: -10.),
+                (x: 0., y: -10.),
+                (x: 0., y: 0.),
+            ],
+            polygon![
+                (x: 0., y: 0.),
+                (x: -10., y: 0.),
+                (x: -10., y: 10.),
+                (x: 0., y: 10.),
+                (x: 0., y: 0.),
+            ]
+        ].into();
+
+        assert_relative_eq!(multipolygon.rotate(-90.), expected, max_relative=1.0);
```

Comment:
I'm curious, does this still pass if you delete the `max_relative=1.0`?

My understanding of `max_relative=1.0` is that you're allowing up to 100% error, which seems unnecessarily large.

Reply:
Removing the `max_relative` does indeed fail:

```
running 1 test
thread 'algorithm::rotate::test::test_rotate_multipolygon_around_centroid' panicked at 'assert_relative_eq!(multipolygon.rotate(-90.), expected, max_relative = 0.1)

    left  = MultiPolygon([Polygon { exterior: LineString([Coordinate { x: 0.0, y: 0.0 }, Coordinate { x: 0.0000000000000006123233995736766, y: -10.0 }, Coordinate { x: 10.0, y: -10.0 }, Coordinate { x: 10.0, y: 0.0000000000000006123233995736766 }, Coordinate { x: 0.0, y: 0.0 }]), interiors: [] }, Polygon { exterior: LineString([Coordinate { x: 0.0, y: 0.0 }, Coordinate { x: -0.0000000000000006123233995736766, y: 10.0 }, Coordinate { x: -10.0, y: 10.0 }, Coordinate { x: -10.0, y: -0.0000000000000006123233995736766 }, Coordinate { x: 0.0, y: 0.0 }]), interiors: [] }])
    right = MultiPolygon([Polygon { exterior: LineString([Coordinate { x: 0.0, y: 0.0 }, Coordinate { x: 10.0, y: 0.0 }, Coordinate { x: 10.0, y: -10.0 }, Coordinate { x: 0.0, y: -10.0 }, Coordinate { x: 0.0, y: 0.0 }]), interiors: [] }, Polygon { exterior: LineString([Coordinate { x: 0.0, y: 0.0 }, Coordinate { x: -10.0, y: 0.0 }, Coordinate { x: -10.0, y: 10.0 }, Coordinate { x: 0.0, y: 10.0 }, Coordinate { x: 0.0, y: 0.0 }]), interiors: [] }])
```

My knowledge of floating point comparison is sketchy enough to where I'm not sure what this a tuning param for this is. It's possible I've just missed something stupid tho, so I'll look more tomorrow.

Reply:
Ignoring the *small* errors for now, a quick glance shows the coords seem to be _way_ off. 

e.g. look at the second coord of the first poly:

```
a: Coordinate { x: 0.0000000000000006123233995736766, y: -10.0 }
b: Coordinate { x: 10.0, y: 0.0 }
```

One thing that could be going on... be aware that the equals check is _not_ a topology check — to be `eq`, the MultiPolygons can't just have the same **shape**, but also their member Polygons must be in the same order, and each members coords must be in the same order.

Maybe something going on there?

Reply:
Thanks for the pointers. I tried to short cut my way to a rotation I thought I could do by hand but I either botched the expected or missed something subtle. I think I'll create a real example and then run it through geos to get the expected values.￼